nix/tests/vm-tests/web-ghc/default.nix

```diff
@@ -0,0 +1,31 @@
+{ pkgs, web-ghc }:
+
+let
+  webghcPort = 8080;
+  webghcHealthEndpoint = "http://localhost:${toString webghcPort}/health";
+  webghcCompileEndpoint = "http://localhost:${toString webghcPort}/runghc";
+  webghcVersionEndpoint = "http://localhost:${toString webghcPort}/version";
+in
+
+pkgs.nixosTest {
+
+  machine = { pkgs, ... }: {
+    nixpkgs.overlays = [ (self: super: { inherit web-ghc; }) ];
+    imports = [ ./module.nix ];
+    services.web-ghc = {
+      enable = true;
+      port = webghcPort;
+    };
+  };
+
+  testScript = { nodes, ... }: ''
+    # fmt: off
+    machine.start()
+    machine.wait_for_unit("web-ghc.service")
```

Comment:
Is there a variable that evaluates to `web-ghc.service` that we could use?

Reply:
no, unfortunately not.